realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -433,22 +435,83 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedRealm_nativeGetTable(JNIEnv
 
 JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedRealm_nativeCreateTable(JNIEnv* env, jclass,
                                                                              jlong shared_realm_ptr,
-                                                                             jstring table_name)
+                                                                             jstring table_name,
+                                                                             jboolean is_pk_table)
 {
     TR_ENTER_PTR(shared_realm_ptr)
 
-    std::string name_str;
+    std::string table_name_str;
     try {
-        JStringAccessor name(env, table_name); // throws
-        name_str = name;
+        table_name_str = JStringAccessor(env, table_name); // throws
         auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
         shared_realm->verify_in_write(); // throws
-        Table* table = LangBindHelper::add_table(shared_realm->read_group(), name); // throws
+        Table* table;
+        auto& group = shared_realm->read_group();
+        if (is_pk_table) {
+            // sync::create_table() will add an extra column for stable ID which is not allowed for pk table.
```

Comment:
If `__0ID` is added first, then the column index for the PK table will be changed which are hard coded everywhere. Object Store is not using sync's APK to create PK table either. https://github.com/realm/realm-object-store/blob/2aa939f1c0a2f849f67f76ec6a736e7dae36c1ea/src/object_store.cpp#L61